testing/playwright/fixtures/helpers/resourceCreationHelpers.ts

```diff
@@ -0,0 +1,289 @@
+      apiVersion: 'v1',
+      kind: 'Secret',
+      metadata: {
+        name: secretName,
+        namespace: MTV_NAMESPACE,
+      },
+      type: 'Opaque',
+      stringData: {
+        url: testProviderData.hostname,
+      },
+    };
+
+    const createdSecret = await resourceManager.createSecret(page, secret);
+    if (!createdSecret) {
+      throw new Error(`Failed to create secret for OVA provider ${testProviderData.name}`);
+    }
+
+    // Create provider with applianceManagement setting
+    const provider: V1beta1Provider = {
+      apiVersion: 'forklift.konveyor.io/v1beta1',
+      kind: 'Provider',
+      metadata: {
+        name: testProviderData.name,
+        namespace: MTV_NAMESPACE,
+      },
+      spec: {
+        type: ProviderType.OVA,
+        url: testProviderData.hostname,
+        secret: {
+          name: secretName,
+          namespace: MTV_NAMESPACE,
+        },
+        settings: {
+          applianceManagement: 'true',
+        },
+      },
+    };
+
+    const createdProvider = await resourceManager.createProvider(page, provider);
+    if (!createdProvider) {
+      throw new Error(`Failed to create OVA provider ${testProviderData.name}`);
+    }
+
+    // Register provider for cleanup
+    resourceManager.addProvider(testProviderData.name, MTV_NAMESPACE);
+  } else {
+    // For non-OVA providers, use the UI
+    const createProviderPage = new CreateProviderPage(page, resourceManager);
+
+    await createProviderPage.navigate();
+    await createProviderPage.create(testProviderData);
+  }
+
+  const provider: TestProvider = {
+    apiVersion: 'forklift.konveyor.io/v1beta1',
+    kind: 'Provider',
+    metadata: {
+      name: testProviderData.name,
+      namespace: MTV_NAMESPACE,
+    },
+  };
+
+  (provider as any).testData = testProviderData;
+  return provider;
+};
+
+export const createPlanWithCustomData = async (
+  page: Page,
+  resourceManager: ResourceManager,
+  options: {
+    sourceProvider: V1beta1Provider;
+    customPlanData?: Partial<ReturnType<typeof createPlanTestData>>;
+  },
+): Promise<TestPlan> => {
+  const { sourceProvider, customPlanData } = options;
+  const defaultPlanData = createPlanTestData({
+    sourceProvider: sourceProvider.metadata!.name!,
+  });
+
+  const testPlanData = customPlanData ? { ...defaultPlanData, ...customPlanData } : defaultPlanData;
+
+  const createWizard = new CreatePlanWizardPage(page, resourceManager);
+  const planDetailsPage = new PlanDetailsPage(page);
+
+  await createWizard.navigate();
+  await createWizard.waitForWizardLoad();
+  await createWizard.fillAndSubmit(testPlanData);
+  await planDetailsPage.verifyPlanTitle(testPlanData.planName);
+  const plan: TestPlan = {
+    apiVersion: 'forklift.konveyor.io/v1beta1',
+    kind: 'Plan',
+    metadata: {
+      name: testPlanData.planName,
+      namespace: MTV_NAMESPACE,
+    },
+    testData: testPlanData,
+  };
+
+  return plan;
+};
+
+export const createPlan = async (
```

Comment:
here too, why not get the createPlanWithCustomData implementation into createPlan? why we need 2 functions?